src/arrayMethodJoin.js

```diff
@@ -3,9 +3,30 @@
 /**
  * Implement method join
  */
+
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    }
+
+    let resultStr;
+
+    if (this[0] !== null) {
+      resultStr = `${this[0]}`;
+    } else {
+      resultStr = '';
+    }
```

Comment:
Could be simplified)
```suggestion
    let resultStr = this[0] !== null ? `${this[0]}` : '';
```